src/app/inventory/advanced-write-actions.ts

```diff
@@ -52,13 +52,14 @@ function hasInsertionCost(defs: D2ManifestDefinitions, plug: DestinyInventoryIte
 
 function canInsertForFree(
   socket: DimSocket,
-  plugItemHash: number,
+  plugItemHash: number | undefined,
```

Comment:
Why would this be able to operate on undefined now?

Reply:
I can revert but it was just an easy way at the time to deal with this plug item hash potentially being undefined, which can happen in the case of artificer mods. I will pull it further up the chain